cms/models.py

```diff
@@ -636,6 +637,26 @@ def save(self, *args, **kwargs):
         self.slug = slugify("{}-{}".format(self.get_parent().id, self.title))
         super().save(*args, **kwargs)
 
+    def get_url_parts(self, request=None):
+        """
+        Override how the url is generated for course/program child pages
+        """
+        # Verify the page is routable
+        url_parts = super().get_url_parts(request=request)
+
+        if not url_parts:
+            return None
+
+        site_id, site_root, parent_path = self.get_parent().specific.get_url_parts(
+            request=request
+        )
+        page_path = ""
+        if WAGTAIL_APPEND_SLASH:
```

Comment:
Can you please add a description over it ? what actually we are doing here.

Reply:
I thought it was obvious, this is pretty much the same as Django's `APPEND_SLASH` meaning the generated URL will have a trailing `/`. I'll add the comment anyway.

Reply:
Done